pkg/util/util.go

```diff
@@ -198,3 +200,17 @@ func InitObjectMeta(obj metav1.Object) {
 		obj.SetAnnotations(map[string]string{})
 	}
 }
+
+// GenerateProxySecret generate random secret key for oauth proxy cookie.
+func GenerateProxySecret() string {
+	const secretLength = 16
+	randString := make([]byte, secretLength)
+	_, err := rand.Read(randString)
+	if err != nil {
+		// If we cannot generate random, return fixed.
+		return "ncNDoqLGrayxXzxTn5ANbOXZp3qXd0LA"
```

Comment:
Return an error instead, and let the caller decide. At the very least, there should be a WARN in the logs about this. It was really bad that I didn't realize that "SECRET" should have been a real random string...